dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EnrollmentAttributeValidationHook.java

```diff
@@ -205,16 +204,7 @@ private Set<MetadataIdentifier> buildTeiAttributes( TrackerBundle bundle,
 
     private MetadataIdentifier getOrgUnitUidFromTei( TrackerBundle bundle, String teiUid )
     {
-
-        final Optional<ReferenceTrackerEntity> reference = bundle.getPreheat().getReference( teiUid );
-        if ( reference.isPresent() )
-        {
-            final Optional<TrackedEntity> tei = bundle.findTrackedEntityByUid( teiUid );
-            if ( tei.isPresent() )
-            {
-                return tei.get().getOrgUnit();
-            }
-        }
-        return null;
+        final Optional<TrackedEntity> tei = bundle.findTrackedEntityByUid( teiUid );
```

Comment:
I think you can inline `tei`. Using some indentation/newlines it should be readable.